qiskit/dagcircuit/dagcircuit.py

```diff
@@ -733,10 +683,10 @@ def _make_pred_succ_maps(self, node):
                 nodes of n.
         """
 
-        pred_map = {e[2]['wire']: self._id_to_node[e[0]] for e in
-                    self._get_multi_graph_in_edges(node._node_id)}
-        succ_map = {e[2]['wire']: self._id_to_node[e[1]] for e in
-                    self._get_multi_graph_out_edges(node._node_id)}
+        pred_map = {e[2]['wire']: e[0] for e in
+                    self._multi_graph.in_edges(node._node_id)}
+        succ_map = {e[2]['wire']: e[1] for e in
+                    self._multi_graph.out_edges(node._node_id)}
```

Comment:
It's been a while since I looked at it so I'm not sure, maybe? But it doesn't really matter because the only place the method is used `pred_map[wire]` and `succ_map[wire]`  gets used the value in the dict gets passed to a retworkx function which wants an index

Reply:
So I did a little more digging the return type change here is fine because this PR also updates all the uses of this function (and `_full_pred_succ_maps`) to assume an index is returned instead of a node object. We were actually doing `pred_map[x]._node_id` in all the places before that were using these dicts, like: https://github.com/Qiskit/qiskit-terra/pull/4386/files#diff-feb434f18758792899804d6e18c57707L1159

Reply:
:+1:, can you add this to the return type info in the docstring?

Reply:
Sure, done in https://github.com/Qiskit/qiskit-terra/pull/4386/commits/38af5bcb637453004fd1ae0c8af994f1df995e9d